src/pages/studyView/table/FusionGenesTable.tsx

```diff
@@ -0,0 +1,310 @@
+import LabeledCheckbox from "../../../shared/components/labeledCheckbox/LabeledCheckbox";
+import MobxPromise from "mobxpromise";
+import {If} from 'react-if';
+import * as _ from 'lodash';
+import classnames from 'classnames';
+import DefaultTooltip from "shared/components/defaultTooltip/DefaultTooltip";
+import FixedHeaderTable from "./FixedHeaderTable";
+import {action, computed, IReactionDisposer, observable, reaction} from "mobx";
+import autobind from 'autobind-decorator';
+import {
+    correctMargin,
+    correctColumnWidth,
+    getFixedHeaderNumberCellMargin,
+    getFixedHeaderTableMaxLengthStringPixel,
+    getFrequencyStr,
+    getQValue
+} from "../StudyViewUtils";
+import {SortDirection} from "../../../shared/components/lazyMobXTable/LazyMobXTable";
+import {DEFAULT_SORTING_COLUMN} from "../StudyViewConfig";
+
+export interface IFusionGenesTablePros {
+    promise: MobxPromise<FusionGenesData>;
+    width: number;
+    height: number;
+    filters: number[];
+    onUserSelection: (value: GeneIdentifier[]) => void;
+    numOfSelectedSamples: number;
+    onGeneSelect: (hugoGeneSymbol: string) => void;
+    selectedGenes: string[];
+}
+
+type FusionGenesTableUserSelectionWithIndex = {
+    entrezGeneId: number;
+    hugoGeneSymbol: string;
+    rowIndex: number;
+}
+
+enum ColumnKey {
+    GENE = 'Gene',
+    NUMBER_MUTATIONS = '# Mut',
+    NUMBER = '#'
+}
+
+class FusionGenesTableComponent extends FixedHeaderTable<MutationCountByGene> {
+}
+
+@observer
+export class FusionGenesTable extends React.Component<IFusionGenesTablePros, {}> {
+    @observable private preSelectedRows: FusionGenesTableUserSelectionWithIndex[] = [];
+    @observable private sortBy: string = DEFAULT_SORTING_COLUMN;
+    @observable private sortDirection: SortDirection;
+    @observable private cellMargin: { [key: string]: number } = {
+        [ColumnKey.GENE]: 0,
+        [ColumnKey.NUMBER_MUTATIONS]: 0,
+        [ColumnKey.NUMBER]: 0
+    };
+
+    private disposers:IReactionDisposer[] = [];
+
+    constructor(props: IFusionGenesTablePros) {
+        super(props);
+        this.disposers.push(
+            reaction(() => this.columnsWidth, () => {
+                this.updateCellMargin();
+            }, {fireImmediately: true})
+        );
+        this.disposers.push(
+            reaction(() => this.props.promise.result, () => {
+                this.updateCellMargin();
+            }, {fireImmediately: true})
+        );
+    }
+
+    componentWillUnmount() {
+        for (const disposer of this.disposers) {
+            disposer();
+        }
+    }
+
+    @computed
+    get columnsWidth() {
+        return {
+            [ColumnKey.GENE]: correctColumnWidth(this.props.width * 0.35),
+            [ColumnKey.NUMBER_MUTATIONS]: correctColumnWidth(this.props.width * 0.25),
+            [ColumnKey.NUMBER]: correctColumnWidth(this.props.width * 0.25)
+        };
+    }
+
+    @autobind
+    @action
+    updateCellMargin() {
```

Comment:
this seems like it could be shared b/c there's nothing implementation specific.  or is there?

Reply:
no longer use updateCellMargin().